kicksaw_integration_utils/aws/sqs/queue.py

```diff
@@ -0,0 +1,339 @@
+            Message.
+
+        Returns
+        -------
+        message_id : str
+            Id of the sent message. Randomly generated by SQS and is
+
+        """
+        logger.debug("Sending message %s to %s", message, self.name)
+        response = self._queue.send_message(MessageBody=message.json())
+        message_id: str = response.get("MessageId")
+        logger.debug(
+            "Successfully sent message %s to %s, SQS Id: %s",
+            message,
+            self.name,
+            message_id,
+        )
+        return message_id
+
+    def send_messages(self, messages: List[PydanticModel]) -> List[bool]:
+        """
+        Send multiple messages to the queue.
+
+        Parameters
+        ----------
+        messages : List[pydantic.BaseModel]
+            List of messages.
+
+        Returns
+        -------
+        List[bool]
+            List of results (True for success, False for failure).
+
+        Raises
+        ------
+        AssertionError
+            Number of results doesn't match number of messages sent.
+
+        """
+        logger.debug(
+            "Sending %d messages to %s in batches of 10",
+            len(messages),
+            self.name,
+        )
+
+        results: List[bool] = []
+        for i in range(0, len(messages), 10):
+            batch = messages[i : i + 10]
+            logger.debug(
+                "Sending batch of %d messages to %s",
+                len(batch),
+                self.name,
+            )
+            response = self._queue.send_messages(
+                Entries=[
+                    {"Id": f"{i+j}", "MessageBody": message.json()}
+                    for j, message in enumerate(batch)
+                ]
+            )
+
+            # Parse response
+            batch_results = sorted(
+                [
+                    *[
+                        {"id": int(entry["Id"]), "success": True}
+                        for entry in response.get("Successful", [])
+                    ],
+                    *[
+                        {"id": int(entry["Id"]), "success": False}
+                        for entry in response.get("Failed", [])
+                    ],
+                ],
+                key=lambda x: x["id"],
+            )
+            assert len(batch_results) == len(batch), (
+                f"this is a bug, AWS returned {len(batch_results):,d} responses, "
+                f"{len(batch):,d} messages were sent"
+            )
+            results.extend([value["success"] for value in batch_results])
+
+        for message, result in zip(messages, results):
+            if not result:
+                logger.warning(
+                    "Failed to send message %s to %s",
+                    message,
+                    self.name,
+                )
+
+        return results
+
+    def receive_messages(
+        self,
+        max_messages: int = 10_000,
+        wait_time_seconds: int = 0,
+        max_poll_attempts: int = 0,
```

Comment:
this isn't _really_ necessary, but I'd almost want to set a maximum value for this since it's so dangerous, like a
```
assert max_poll_attempts < 10000
```
or something

Reply:
This was a good comment because I found a bug that would cause execution continue until `max_poll_attempts` even if queue is actually empty (returning 0 messages), updated in new commit.